mock/ecs_client_test.go

```diff
@@ -64,7 +64,7 @@ func TestECSClient(t *testing.T) {
 			tctx, tcancel := context.WithTimeout(ctx, defaultTestTimeout)
 			defer tcancel()
 
-			cleanupECSAndSecretsManagerCache()
+			resetECSClustersAndSecretsManagerCache()
```

Comment:
Just curious why do we want different resetting logic here?

Reply:
This is specifically because [L57](https://github.com/evergreen-ci/cocoa/pull/66/files#diff-2c182caa772df3b10e8de552832ce5d47f9636bbd677cc85b9ebe6bd87fa0592R57) registers an ECS task definition to be reused between all test cases in `testcase.ECSClientRegisteredTaskDefinitionTests`. If the task definitions were cleaned up, the original registered task definition passed on [L62](https://github.com/evergreen-ci/cocoa/pull/66/files#diff-2c182caa772df3b10e8de552832ce5d47f9636bbd677cc85b9ebe6bd87fa0592R62) would no longer be valid. However, since this is just making API calls to a fake in-memory ECS service, I can change the way `testcase.ECSClientRegisteredTaskDefinitionTests` is written so that we don't need this special cleanup function.